pymbar/fes.py

```diff
@@ -692,8 +692,9 @@ def _generate_fes_kde(self, b, x_n, w_n):
             params = self.kde.get_params()  # pylint: disable=access-member-before-definition
             kde.set_params(**params)
         else:
-            kde = self.kde
-        kde.fit(x_n, sample_weight=self.w_n)
+            kde = self.kde  # use these new weights for the KDE
+            w_n = self.w_n
```

Comment:
I actually can't remember if this was 100% necessary to get updated weights . . .